st2common/st2common/services/triggers.py

```diff
@@ -46,6 +49,27 @@ def get_trigger_db_given_type_and_params(type=None, parameters=None):
 
         trigger_db = trigger_dbs[0] if len(trigger_dbs) > 0 else None
 
+        # Work around for cron-timer when in some scenarios finding an object fails when Python
+        # value types are unicode :/
+        if not trigger_db and six.PY2 and type == CRON_TIMER_TRIGGER_REF and parameters:
+            non_unicode_literal_parameters = {}
+            for key, value in six.iteritems(parameters):
+                key = key.encode('utf-8')
+                value = value.encode('utf-8')
+                non_unicode_literal_parameters[key] = value
+            parameters = non_unicode_literal_parameters
+
+            trigger_dbs = Trigger.query(type=type,
+                                        parameters=non_unicode_literal_parameters).no_cache()
+
+            # Note: We need to directly access the object, using len or accessing the query set
+            # twice won't work - there seems to bug a bug with cursor where accessing it twice
+            # will throw an exception
+            try:
+                trigger_db = trigger_dbs[0]
```

Comment:
This is also terrible, but if I try to access the QuerySet twice (perform the same check as above or similar), it throws:

```python
  File "/data/stanley/virtualenv/local/lib/python2.7/site-packages/pymongo/cursor.py", line 660, in __getitem__
    raise IndexError("no such item for Cursor instance")
```

Basically, it looks like that accessing query set in any kind of way (even just by checking the length) causes cursor to advance to the very end. This is another, what looks like a bizarre bug with cursor handling in pymongo / mongoengine.